pkg/discovery/helper.go

```diff
@@ -109,7 +109,11 @@ func (h *helper) Refresh() error {
 
 	preferredResources, err := h.discoveryClient.ServerPreferredResources()
 	if err != nil {
-		return errors.WithStack(err)
+		if discovery.IsGroupDiscoveryFailedError(err) {
+			h.logger.Warnf("Failed to discover some groups: %v", err.(*discovery.ErrGroupDiscoveryFailed).Groups)
```

Comment:
I think it might be a little easier to read the log if instead of logging everything in one line, we loop through the groups:

```
for groupVersion, err := range err.(*discovery.ErrGroupDiscoveryFailed).Groups {
    // h.logger.WithError(err).Warnf(...)
}
```